pkg/storegateway/batch_series_test.go

```diff
@@ -734,6 +763,9 @@ func TestPreloadingBatchSet(t *testing.T) {
 		// Cancel the context. Do NOT call Next() after canceling the context.
 		cancelCtx()
 	})
+
+	// Give goroutines time to stop so that they won't be reported as leaking.
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Is this really needed? `leaktest` should retry several times if it found leaking goroutines before failing.

Reply:
I didn't know it! I checked the code and you're right. Removed.